### DISCUSSION TITLE
"Columns" dropdown not working and styling incorrect

### DISCUSSION THREAD

**Discussion by nesben** [rappasoft/laravel-livewire-tables#1183] [General]
Hello! I've been playing around with livewire tables, and I can't seem to get it working...

Whenever i click inside the "Search" input, i receive an error (if the field is empty)
![image](https://user-images.githubusercontent.com/26848437/236415522-9231d264-af27-4151-86e1-beb1206a4cbc.png)

Furthermore, the "Columns" dropdown field doesn't open when clicking, nor does it error in the console.

Also, the styling seems quite messed up... some fields have styling, others don't, and I can't seem to figure out why...
![image](https://user-images.githubusercontent.com/26848437/236415788-ec4f2afb-0bcb-4952-8939-0b808fd02984.png)

Any helps is appreciated 😄 

Here's my code:

**resources/js/app.js**
```js
import './bootstrap';
import Alpine from "alpinejs";
import focus from '@alpinejs/focus';
import '../css/app.css';

window.Alpine = Alpine;
window.Alpine.plugin(focus);
Alpine.start();
```

**resources/views/tenant/layout/layout.blade.php**
```blade
<!DOCTYPE html>
<html lang="{{ str_replace('_', '-', app()->getLocale()) }}">

<head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">

    <title>Test</title>

    @vite('resources/js/app.js')

    @livewireStyles
</head>

<body>
    <main class="flex h-screen bg-white">
        <aside class="bg-white w-[15%] h-screen">
            <ul class="list-none mt-[50px]">
                @foreach ($tenant_routes as $route)
                    <li class="@active(strtolower($route->name)) mb-3 flex items-center w-full p-2 ps-10 text-gray-900 group">
                        <a href="{{ route(strtolower($route->name), ['tenant' => tenant()]) }}">{{ $route->name }}</a>
                    </li>
                @endforeach
            </ul>
        </aside>

        <section class="flex-1 p-[50px] rounded-tr-50 rounded-tl-[50px] bg-gray-100">
            <!-- Main Content -->
            <h2 id='page-title' class="font-bold
            text-2xl mb-12">{{ $title }}</h2>
            <div id='page-content' class="page-content">
                @yield('content')
            </div>
        </section>
    </main>

    @livewireScripts
    <script>
        window.Livewire.connection.headers = {
            ...window.Livewire.connection.headers,
            'X-Tenant': '{{ tenant()->id }}'
        };
    </script>

</body>

</html>
```

**resources/css/app.css**
```css
@import "@fortawesome/fontawesome-free/css/all.css";

@tailwind base;
@tailwind components;
@tailwind utilities;

/* Reset CSS */
* {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
    font-family: 'Nunito', sans-serif;
    overflow: hidden;
}

[x-cloak] {
    display: none !important;
}
```

**tailwind.config.js**
```js
/** @type {import('tailwindcss').Config} */
module.exports = {
  mode: 'jit',
  darkMode: 'class', // or 'media' or 'class'
  content: [
    "./resources/**/*.blade.php",
    "./resources/**/*.js",
    "./resources/**/*.vue",
    "./vendor/rappasoft/laravel-livewire-tables/resources/views/**/*.blade.php",
  ],
  theme: {
    extend: {},
  },
}
```

**resources/views/tenant/sales/invoices.blade.php**
```blade
@extends('tenant.layout.layout', ['title' => 'Test2'])

@section('content')
    <div>
        <div class="grid gap-6 mb-8 md:grid-cols-2 xl:grid-cols-4">
            <x-tenant.card title="Card Title" content="Card Content" type="Card Type" size="w-96" icon='test' />
            <x-tenant.card title="Card Title" content="Card Content" type="Card Type" size="w-96" icon='test' />
            <x-tenant.card title="Card Title" content="Card Content" type="Card Type" size="w-96" icon='test' />
        </div>


        <div>
            <livewire:tenant.invoices-table />
        </div>
    </div>
@endsection
```

**vite.config.js**
```js
import { defineConfig } from 'vite';
import laravel from 'laravel-vite-plugin';

export default defineConfig({
    plugins: [
        laravel({
            input: ['resources/js/app.js'],
            refresh: true,
        }),
    ],
});
```

**Comment by lrljoe** [comment]
Can you confirm:
which package version you're using
Tailwind version
Laravel Version
Livewire Version
Alpine version

You need to include the tailwind forms plugins etc to maintain styling in your Tailwind config file

**Reply by nesben** [reply]
**Version list:**
laravel-livewire-tables: v2.12
Tailwind: v3.3.2
Laravel: v10.9.0
Livewire: v2.12
Alpine: v3.12.0
Vite: v4.0.0

I added @tailwindcss/forms plugin in my tailwind file, and it helped slightly with the view, but button press on "Columns" still doesn't work, nor does styling look exactly right...
![image](https://user-images.githubusercontent.com/26848437/236527700-c0ac430c-858a-4963-bdd3-af8d82b9e0c9.png)

**Reply by lrljoe** [reply]
So my tailwind.config.js for my Table Demo environment looks like this.

This only has relevant pieces in for this package, and producing the demo environment.

```js
/** @type {import('tailwindcss').Config} */

const defaultTheme = require('tailwindcss/defaultTheme');
const colors = require('tailwindcss/colors')

module.exports = {
    mode: 'jit',
    plugins: [
        require('@tailwindcss/forms'),
        require('@tailwindcss/typography'),
        require('@tailwindcss/aspect-ratio'),
        require('@tailwindcss/line-clamp')
    ],
    darkMode: 'class',
    purge: {
        content: [
            './vendor/rappasoft/laravel-livewire-tables/resources/views/**/*.blade.php',
            './storage/framework/views/*.php',
            './resources/views/**/*.blade.php',
        ],
    },
    theme: {
        extend: {
            fontFamily: {
                sans: ['Inter var', ...defaultTheme.fontFamily.sans],
            },
        },
    },
    variants: {
        extend: {
            backgroundColor: ['responsive', 'dark', 'checked', 'disabled', 'hover', 'focus', 'active', 'even', 'odd'],
            textColor: ['disabled'],
            opacity: ['dark'],
            overflow: ['hover'],
        }
    }
};
```

Try running an "npm run build" so that you can see any compilation errors in real-time, as you may be missing a JS package locally.

**Reply by nesben** [reply]
After using your tailwind config, I didn't receive any errors, and the "columns" button is still not responding to button clicks... what can it be?
![image](https://user-images.githubusercontent.com/26848437/236559983-6dcfd39f-56f9-4176-a639-14cb992a0c04.png)

**Reply by lrljoe** [reply]
Firstly, apologies for being absent for so long!

So if you've matched my config, it's a little unusual that it's not working at all.

Feel free to ping me a link to your site nesben@test.lowerrocklabs.com and I will take a look.

I suspect that something is missing from your package.json - are you able to share your package.json and package.lock please?

Just to add - there are a couple of packages that will help you reduce the size of your woff2/ttf files, ping me a message joe@lowerrocklabs.com for some guidance on that as it's well outside the scope of this package

Thanks,

Joe,

**Reply by lrljoe** [reply]
Did you manage to get this sorted in the end?

**Reply by nesben** [reply]
@lrljoe Hi, no I didn't get it fixed.

Though, I have some more information regarding the behaviour...

I tried adding filters, and the dropdown experiences the same issue - it doesn't show when i click on it.

It seems like a styling problem with tailwind possibly... Because when i click the buttons, they change the "aria-expanded" to "true".
![image](https://github.com/rappasoft/laravel-livewire-tables/assets/26848437/e8158d6d-7836-457b-b133-2434469937d2)

**Reply by lrljoe** [reply]
Are you getting any errors in your console if you open devtools in your browser and click on the drop-down?

If you have published the views previously then try removing the published views too

Can you paste your vite config file in too please

**Reply by nesben** [reply]
@lrljoe
I apologise for the late reply...

I haven't published the views.

When clicking on the dropdown (both filters and columns dropdown) I don't get any errors. Console is empty. Though, when i click on the search input field i get the following error:
![image](https://github.com/rappasoft/laravel-livewire-tables/assets/26848437/e29ade18-db1c-4d26-927d-576ff9dd6154)

But that error doesn't prevent me from searching. Searching with the input works fine, but the error shows up when the input is empty and I click on it.

Heres my vite config:
```
import { defineConfig } from 'vite';
import laravel from 'laravel-vite-plugin';

export default defineConfig({
    plugins: [
        laravel({
            input: ['resources/js/app.js'],
            refresh: true,
        }),
    ],
});
```

**Reply by nesben** [reply]
@lrljoe I believe it's an issue with loading Alpine.js. I'm importing alpinejs in my `app.js` which is imported in the main layout file. The alpinejs stuff in my main layout file works fine, but not the table for some reason...

**Reply by lrljoe** [reply]
are you still having an issue? apologies for not coming back sooner...

**Comment by itsmill3rtime** [comment]
you need to add the additional tailwind packages and then add to tailwind.config.js

`yarn add @tailwindcss/typography @tailwindcss/forms --dev`


and then inside tailwind.config.js

import forms from '@tailwindcss/forms';
import typography from '@tailwindcss/typography';

and within the plugins: [ ] array at the bottom add them
```
  plugins: [
        forms,
        typography,
    ],
```

**Comment by lrljoe** [comment]
Please don't reply to questions from years ago!
